pkg/manifests/manifests_test.go

```diff
@@ -779,26 +780,29 @@ func TestPrometheusOperatorConfiguration(t *testing.T) {
 		t.Fatalf("expected node selector to be master, got %q", got)
 	}
 
-	expectedPromOpImage := "docker.io/openshift/origin-prometheus-operator:latest"
-	resPromOpImage := d.Spec.Template.Spec.Containers[0].Image
-	if resPromOpImage != expectedPromOpImage {
-		t.Fatalf("Configuring the Prometheus Operator image failed, expected: %v, got %v", expectedPromOpImage, resPromOpImage)
-	}
-
 	prometheusReloaderFound := false
 	prometheusWebTLSCipherSuitesArg := ""
 	prometheusWebTLSVersionArg := ""
-	for i := range d.Spec.Template.Spec.Containers[0].Args {
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") {
-			prometheusReloaderFound = true
-		}
-
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSCipherSuitesFlag) {
-			prometheusWebTLSCipherSuitesArg = d.Spec.Template.Spec.Containers[0].Args[i]
-		}
+	kubeRbacProxyTLSCipherSuitesArg := ""
+	kubeRbacProxyMinTLSVersionArg := ""
+	for _, container := range d.Spec.Template.Spec.Containers {
+		switch container.Name {
+		case "prometheus-operator":
+			if container.Image != "docker.io/openshift/origin-prometheus-operator:latest" {
+				t.Fatalf("%s image incorrectly configured", container.Name)
+			}
 
-		if strings.HasPrefix(d.Spec.Template.Spec.Containers[0].Args[i], PrometheusOperatorWebTLSMinTLSVersionFlag) {
-			prometheusWebTLSVersionArg = d.Spec.Template.Spec.Containers[0].Args[i]
+			if getContainerArgValue(d.Spec.Template.Spec.Containers, PrometheusConfigReloaderFlag+"docker.io/openshift/origin-prometheus-config-reloader:latest") != "" {
+				prometheusReloaderFound = true
+			}
+			prometheusWebTLSCipherSuitesArg = getContainerArgValue(d.Spec.Template.Spec.Containers, PrometheusOperatorWebTLSCipherSuitesFlag)
+			prometheusWebTLSVersionArg = getContainerArgValue(d.Spec.Template.Spec.Containers, PrometheusOperatorWebTLSMinTLSVersionFlag)
+		case "kube-rbac-proxy":
+			if container.Image != "docker.io/openshift/origin-kube-rbac-proxy:latest" {
+				t.Fatalf("%s image incorrectly configured", container.Name)
+			}
+			kubeRbacProxyTLSCipherSuitesArg = getContainerArgValue(d.Spec.Template.Spec.Containers, KubeRbacProxyTLSCipherSuitesFlag)
```

Comment:
Isn't this going to get the value for any container, not the just for kube-rbac-proxy?

Reply:
correct

Reply:
But the variable says `kubeRbacProxyTLSCipherSuitesArg`. It also means that the tests even will pass if we set the argument in the wrong container.

Reply:
since we are trying get only argument value for `KubeRbacProxyTLSCipherSuitesFlag variable is named as `kubeRbacProxyTLSCipherSuitesArg`. Test will not pass since we compare expected value and this value. I had issue initially since it was not extracting argument correctly but once I fixed the function it was matching with expected

Reply:
This line https://github.com/openshift/cluster-monitoring-operator/pull/1466/files#diff-b388a528a3665a46f4dfa8eb4a8c02de116acb8f85e4611f12f11dc337d7f7b5R826

Reply:
The value of `KubeRbacProxyTLSCipherSuitesFlag` is `--tls-cipher-suites=`. The prometheus adapter also uses the same arg name[1]. Furthermore, the newly defined helper function is called `getContainerArgValue` which indicates that it will get the argument value for a specific container. 

I would suggest that we add the container name as an argument to the function and resolve any ambiguity and potential misuses. It will also make the function much more reusable because it will be safe to use it to get the arg value for individual containers.

[1] https://github.com/openshift/cluster-monitoring-operator/blob/361da15b129da47a19ef4f2ae7c5b28eebabdd8c/pkg/manifests/manifests.go#L261

Reply:
Ok that sounds good. I will update

Reply:
updated. Please review :)